test/spv-node-sync-test.js

```diff
@@ -0,0 +1,233 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const consensus = require('../lib/protocol/consensus');
+const FullNode = require('../lib/node/fullnode');
+const SPVNode = require('../lib/node/spvnode');
+
+const ports = {
+  full: {
+    p2p: 49331,
+    node: 49332,
+    wallet: 49333
+  },
+  spv: {
+    p2p: 49431,
+    node: 49432,
+    wallet: 49433
+  }
+};
+
+const node = new FullNode({
+  network: 'regtest',
+  workers: true,
+  listen: true,
+  bip37: true,
+  port: ports.full.p2p,
+  httpPort: ports.full.node,
+  maxOutbound: 1,
+  seeds: [],
+  memory: true,
+  plugins: [require('../lib/wallet/plugin')],
+  env: {
+    'BCOIN_WALLET_HTTP_PORT': (ports.full.wallet).toString()
+  }
+});
+
+const spvnode = new SPVNode({
+  network: 'regtest',
+  workers: true,
+  listen: true,
+  port: ports.spv.p2p,
+  httpPort: ports.spv.node,
+  maxOutbound: 1,
+  seeds: [],
+  nodes: [`127.0.0.1:${ports.full.p2p}`],
+  memory: true,
+  plugins: [require('../lib/wallet/plugin')],
+  env: {
+    'BCOIN_WALLET_HTTP_PORT': (ports.spv.wallet).toString()
+  }
+});
+
+const chain = node.chain;
+const miner = node.miner;
+const {wdb} = node.require('walletdb');
+const {wdb: spvwdb} = spvnode.require('walletdb');
+
+let wallet = null;
+let spvwallet = null;
+let spvaddr = null;
+let tip1 = null;
+let tip2 = null;
+
+async function mineBlock(tip) {
+  const job = await miner.createJob(tip);
+  return await job.mineAsync();
+}
+
+async function event(obj, name) {
+  return new Promise((resolve) => {
+    obj.once(name, resolve);
+  });
+}
+
+describe('SPV Node Sync', function() {
+  this.timeout(10000);
+
+  if (process.browser)
+    return;
+
+  const cbMaturity = consensus.COINBASE_MATURITY;
+  before(async () => {
+    consensus.COINBASE_MATURITY = 0;
+    await node.open();
+    await spvnode.open();
+    await node.connect();
+    await spvnode.connect();
+    await spvnode.startSync();
+  });
+
+  after(async () => {
+    consensus.COINBASE_MATURITY = cbMaturity;
```

Comment:
Unfortunately, this won't be restored in the case that any of the `*.open()` fails, in the `before` hook. I can check to verify. This could then have a side affect that tests following it would have this value changed, so it's best to be careful when modifying these global values.

I'll think about some alternatives, a current one is to generate enough blocks initially for the test to be able to spend (if necessary), this doesn't take too much time, but it may be preferable to not wait?

Reply:
One way to mitigate this would be to assert that the value is the expected value in any tests that depend on the `COINBASE_MATURITY` so that if it is mutated, it will cause tests to fail. Good thinking that mutating this could impact other tests

Reply:
That could get unwieldy, as in some cases there would need to be assertions on _all_ consensus properties.

Reply:
This could be guarded against in the test framework by [running each test in a new child process](https://github.com/braydonf/chainbeacon/blob/master/test/bootstrap.js#L71-L86), and any such modifications would not leak into other tests. Would need to look into that more.

Reply:
Fixed in latest commit, mining 100 blocks is fine, don't have to change the rules.
I did have to add a wallet rescan though otherwise the script will try to proceed with spending coins before the wallet is caught up to the chain.